gnucash/gnome-utils/gnc-report-combo.c

```diff
@@ -0,0 +1,595 @@
+        gchar *guid;
+        gtk_tree_model_get (model, &iter, RC_GUID, &guid, -1);
+
+        if (g_strcmp0 (priv->active_report_guid, guid) == 0)
+        {
+            gtk_combo_box_set_active_iter (GTK_COMBO_BOX(priv->combo), &iter);
+            g_free (guid);
+            found = TRUE;
+            break;
+        }
+        g_free (guid);
+        valid_iter = gtk_tree_model_iter_next (model, &iter);
+    }
+
+    if (!found)
+    {
+        GtkTreeIter iter;
+        QofBook *book = gnc_get_current_book ();
+        gchar *tmp;
+
+        if (priv->active_report_name)
+            tmp = g_strdup (priv->active_report_name);
+        else
+            tmp = g_strdup (_("Selected Report is Missing"));
+
+        gtk_list_store_prepend (GTK_LIST_STORE(model), &iter);
+        gtk_list_store_set (GTK_LIST_STORE(model), &iter,
+                            RC_NAME, tmp,
+                            RC_GUID, priv->active_report_guid,
+                            RC_MISSING, TRUE,
+                            -1);
+
+        g_free (tmp);
+        gtk_combo_box_set_active_iter (GTK_COMBO_BOX(priv->combo), &iter);
+    }
+    return found;
+}
+
+/********************************************************************
+ * update_invoice_list
+ *
+ * this procedure does the real work of displaying a sorted list of
+ * available invoice reports
+ ********************************************************************/
+static void
+update_report_list (GncReportCombo *grc)
+{
+    GncReportComboPrivate *priv = GET_PRIVATE(grc);
+    GtkTreeModel *model = gtk_combo_box_get_model (GTK_COMBO_BOX(priv->combo));
+
+    SCM template_menu_name = scm_c_eval_string ("gnc:report-template-menu-name/report-guid");
+    SCM get_rpt_guids = scm_c_eval_string (priv->rpt_guids);
+    SCM reportlist;
+    SCM rpt_guids;
+
+    gtk_tree_sortable_set_sort_column_id (GTK_TREE_SORTABLE(model),
+                                          RC_NAME, GTK_SORT_ASCENDING);
+
+    gtk_list_store_clear (GTK_LIST_STORE(model));
+
+    if (!scm_is_procedure (get_rpt_guids))
+        return;
+
+    reportlist = scm_call_0 (get_rpt_guids);
+    rpt_guids = reportlist;
+
+    if (scm_is_list (rpt_guids))
+    {
+        int i;
+        GtkTreeIter iter;
+
+        for (i = 0; !scm_is_null (rpt_guids); i++)
+        {
+            gchar *guid_str = scm_to_utf8_string (SCM_CAR(rpt_guids));
+            gchar *name = gnc_scm_to_utf8_string (scm_call_2(template_menu_name,
+                                                  SCM_CAR(rpt_guids), SCM_BOOL_F));
+
+            gtk_list_store_append (GTK_LIST_STORE(model), &iter);
+            gtk_list_store_set (GTK_LIST_STORE(model), &iter,
+                                RC_NAME, name,
+                                RC_GUID, guid_str,
+                                RC_MISSING, FALSE,
+                                -1);
+            g_free (name);
+            g_free (guid_str);
+
+            rpt_guids = SCM_CDR(rpt_guids);
+        }
+    }
```

Comment:
Most of this goes in `gnc_invoice_report_list` and here you iterate over the list and stuff the members into the list store.